diff --git a/src/rddl/competition/Server.java b/src/rddl/competition/Server.java
index c447755..b7a8099 100755
--- a/src/rddl/competition/Server.java
+++ b/src/rddl/competition/Server.java
@@ -56,7 +56,6 @@ public class Server implements Runnable {
 	public static final boolean SHOW_MSG = false;
 	public static final boolean SHOW_TIMING = false;
 	
-	private static final String LOG_FILE = "rddl";
 	/**
 	 * following is XML definitions
 	 */
@@ -117,12 +116,14 @@ public class Server implements Runnable {
 	private RDDL rddl = null;
 	private static int ID = 0;
 	private static int DEFAULT_NUM_ROUNDS = 30;
-    private static long DEFAULT_TIME_ALLOWED = 1080000; // milliseconds = 18 minutes
+	private static long DEFAULT_TIME_ALLOWED = 1080000; // milliseconds = 18 minutes
+	private static boolean USE_TIMEOUT = false;
 	public int port;
 	public int id;
 	public String clientName = null;
 	public String requestedInstance = null;
 	public RandomDataGenerator rand;
+	private static String logFile = "rddl";
 	
 	public State      state;
 	public INSTANCE   instance;
@@ -145,10 +146,10 @@ public class Server implements Runnable {
 		ArrayList<RDDL> rddls = new ArrayList<RDDL>();
 		int port = PORT_NUMBER;
 		if ( args.length < 1 ) {
-			System.out.println("usage: rddlfilename-or-dir (optional) portnumber num-rounds random-seed state-viz-class-name");
+			System.out.println("usage: rddlfilename-or-dir (optional) portnumber num-rounds log-folder random-seed state-viz-class-name");
 			System.out.println("\nexample 1: Server rddlfilename-or-dir");
 			System.out.println("example 2: Server rddlfilename-or-dir 2323");
-			System.out.println("example 3: Server rddlfilename-or-dir 2323 100 0 rddl.viz.GenericScreenDisplay");
+			System.out.println("example 3: Server rddlfilename-or-dir 2323 100 experiments/experiment23/ 0 rddl.viz.GenericScreenDisplay");
 			System.exit(1);
 		}
 				
@@ -163,14 +164,17 @@ public class Server implements Runnable {
 			if (args.length > 2) {
 				DEFAULT_NUM_ROUNDS = Integer.valueOf(args[2]);
 			}
+                        if (args.length > 3) {
+                            logFile = args[3] + "/rddl";
+                        }
 			int rand_seed = -1;
-			if ( args.length > 3) {
-				rand_seed = Integer.valueOf(args[3]);
+			if ( args.length > 4) {
+				rand_seed = Integer.valueOf(args[4]);
 			} else {
 				rand_seed = DEFAULT_SEED;
 			}
-			if (args.length > 4) {
-				state_viz = (StateViz)Class.forName(args[4]).newInstance();
+			if (args.length > 5) {
+				state_viz = (StateViz)Class.forName(args[5]).newInstance();
 			}
 			System.out.println("RDDL Server Initialized");
 			while (true) {
@@ -356,7 +360,7 @@ public class Server implements Runnable {
 					//        an early round end
 					// TODO: check that this works
 					round_elapsed_time = (System.currentTimeMillis() - start_round_time);
-					OUT_OF_TIME = session_elapsed_time + round_elapsed_time > timeAllowed;
+					OUT_OF_TIME = session_elapsed_time + round_elapsed_time > timeAllowed && USE_TIMEOUT;
 				}
 				accum_total_reward += accum_reward;
 				session_elapsed_time += round_elapsed_time;
@@ -397,7 +401,7 @@ public class Server implements Runnable {
 	}
 	
 	public void writeToLog(String msg) throws IOException {
-		BufferedWriter bw = new BufferedWriter(new FileWriter(LOG_FILE + "-" + this.port + ".log" , true));
+		BufferedWriter bw = new BufferedWriter(new FileWriter(logFile + "-" + this.port + ".log" , true));
 		bw.write(msg);
 		bw.newLine();
 		bw.flush();
